CI build job lyrical/testing failed!

An automated scheduled build failed on `master`: https://github.com/ros-controls/ros2_control/actions/runs/26997196184

```

2026-05-29T06:01:50.5884076Z   1 package had test failures: controller_manager
2026-05-29T06:01:50.5884589Z ##[endgroup]
2026-05-29T06:01:50.5884954Z [32m'run_target_test' returned with code '0' after 9 min 12 sec[0m
2026-05-29T06:01:50.5884959Z 
2026-05-29T06:01:50.5886015Z [1m$ ( source /home/runner/work/ros2_control/ros2_control/.work/upstream_ws/install/setup.bash && cd /home/runner/work/ros2_control/ros2_control/.work/target_ws && colcon test-result --verbose; )[0m
2026-05-29T06:01:50.5886480Z build/controller_manager/Testing/20260529-0553/Test.xml: 16 tests, 0 errors, 1 failure, 0 skipped
2026-05-29T06:01:50.5886639Z - test_ros2_control_node
2026-05-29T06:01:50.5886785Z   <<< failure message
2026-05-29T06:01:50.5887008Z     -- run_test.py: extra environment variables:
2026-05-29T06:01:50.5887186Z      - PYTHONDONTWRITEBYTECODE=1
2026-05-29T06:01:50.5887857Z     -- run_test.py: invoking following command in '/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/controller_manager':
2026-05-29T06:01:50.5890754Z      - /usr/bin/python3 -u -m pytest /home/runner/work/ros2_control/ros2_control/.work/target_ws/src/ros2_control/controller_manager/test/test_ros2_control_node_launch.py -o cache_dir=/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/controller_manager/ament_cmake_pytest/test_ros2_control_node/.cache --junit-xml=/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/controller_manager/test_results/controller_manager/test_ros2_control_node.xunit.xml --junit-prefix=controller_manager
2026-05-29T06:01:50.5890932Z     ============================= test session starts ==============================
2026-05-29T06:01:50.5891184Z     platform linux -- Python 3.14.4, pytest-9.0.2, pluggy-1.6.0
2026-05-29T06:01:50.5891439Z     cachedir: build/controller_manager/ament_cmake_pytest/test_ros2_control_node/.cache
2026-05-29T06:01:50.5891636Z     rootdir: /home/runner/work/ros2_control/ros2_control/.work/target_ws
2026-05-29T06:01:50.5892455Z     plugins: ament_flake8-0.20.6, ament_copyright-0.20.6, ament_xmllint-0.20.6, ament_lint-0.20.6, launch_testing_ros-0.29.8, launch_testing-3.9.7, ament_pep257-0.20.6, ament_mypy-0.20.6, typeguard-4.4.4, repeat-0.9.4, cov-5.0.0, colcon-core-0.20.1, rerunfailures-16.1
2026-05-29T06:01:50.5892557Z     collected 1 item
2026-05-29T06:01:50.5892649Z     
2026-05-29T06:01:50.5892902Z     ../../src/ros2_control/controller_manager/test/test_ros2_control_node_launch.py F [100%]
2026-05-29T06:01:50.5892999Z     
2026-05-29T06:01:50.5893151Z     =================================== FAILURES ===================================
2026-05-29T06:01:50.5893361Z     _________________ launch tests: test_ros2_control_node_launch __________________
2026-05-29T06:01:50.5893450Z     
2026-05-29T06:01:50.5893697Z     ============================================================================================================================================================================
2026-05-29T06:01:50.5893899Z     FAIL: test_ros2_control_node_launch.TestShutdown.test_exit_codes
2026-05-29T06:01:50.5894274Z     ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
2026-05-29T06:01:50.5894390Z     Traceback (most recent call last):
2026-05-29T06:01:50.5895004Z       File "/home/runner/work/ros2_control/ros2_control/.work/target_ws/src/ros2_control/controller_manager/test/test_ros2_control_node_launch.py", line 119, in test_exit_codes
2026-05-29T06:01:50.5895172Z         launch_testing.asserts.assertExitCodes(proc_info)
2026-05-29T06:01:50.5895292Z         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
2026-05-29T06:01:50.5895970Z       File "/opt/ros/lyrical/lib/python3.14/site-packages/launch_testing/asserts/assert_exit_codes.py", line 62, in assertExitCodes
2026-05-29T06:01:50.5896236Z         assert info.returncode in allowable_exit_codes, 'Proc {} exited with code {}'.format(
2026-05-29T06:01:50.5896350Z                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-05-29T06:01:50.5896496Z     AssertionError: Proc spawner-3 exited with code -2
2026-05-29T06:01:50.5896689Z     ----------------------------- Captured stdout call -----------------------------
2026-05-29T06:01:50.5896995Z     [INFO] [launch]: All log files can be found below /root/.ros/log/2026-05-29-06-01-44-576975-f74b4e5bd40d-15056
2026-05-29T06:01:50.5897154Z     [INFO] [launch]: Default logging verbosity is set to INFO
2026-05-29T06:01:50.5897334Z     [INFO] [robot_state_publisher-1]: process started with pid [15068]
2026-05-29T06:01:50.5897509Z     [INFO] [ros2_control_node-2]: process started with pid [15069]
2026-05-29T06:01:50.5897657Z     [INFO] [spawner-3]: process started with pid [15070]
2026-05-29T06:01:50.5897933Z     [robot_state_publisher-1] [INFO] [1780034504.673059191] [robot_state_publisher]: Robot initialized
2026-05-29T06:01:50.5898358Z     [ros2_control_node-2] [INFO] [1780034504.685040776] [controller_manager]: Using Steady (Monotonic) clock for triggering controller manager cycles.
2026-05-29T06:01:50.5898753Z     [ros2_control_node-2] [INFO] [1780034504.695859963] [controller_manager]: Subscribing to '/robot_description' topic for robot description.
2026-05-29T06:01:50.5899078Z     [ros2_control_node-2] [INFO] [1780034504.696000874] [controller_manager]: update rate is 100 Hz
2026-05-29T06:01:50.5899372Z     [ros2_control_node-2] [INFO] [1780034504.696014584] [controller_manager]: Overruns handling is : enabled
2026-05-29T06:01:50.5899767Z     [ros2_control_node-2] [INFO] [1780034504.696022596] [controller_manager]: Spawning controller_manager RT thread with scheduler priority: 50
2026-05-29T06:01:50.5900752Z     [ros2_control_node-2] [WARN] [1780034504.696160482] [controller_manager]: Could not enable FIFO RT scheduling policy: with error number <1>(Operation not permitted). See [https://control.ros.org/master/doc/ros2_control/controller_manager/doc/userdoc.html] for details on how to enable realtime scheduling.
2026-05-29T06:01:50.5901072Z     [ros2_control_node-2] [INFO] [1780034504.697157833] [controller_manager]: Received robot description from topic.
2026-05-29T06:01:50.5901369Z     [ros2_control_node-2] [INFO] [1780034504.697199695] [controller_manager]: Enforcing command limits is enabled...
2026-05-29T06:01:50.5901937Z     [ros2_control_node-2] [INFO] [1780034504.701044561] [controller_manager]: Using JointLimiter for joint 'joint2' in hardware 'TestSystemComponent' : '  has position limits: true [-3.14159, 3.14159]
2026-05-29T06:01:50.5902091Z     [ros2_control_node-2]   has velocity limits: true [0.2]
2026-05-29T06:01:50.5902258Z     [ros2_control_node-2]   has acceleration limits: false [nan]
2026-05-29T06:01:50.5902410Z     [ros2_control_node-2]   has deceleration limits: false [nan]
2026-05-29T06:01:50.5902551Z     [ros2_control_node-2]   has jerk limits: false [nan]
2026-05-29T06:01:50.5902690Z     [ros2_control_node-2]   has effort limits: true [0.1]
2026-05-29T06:01:50.5902831Z     [ros2_control_node-2]   angle wraparound: false'
2026-05-29T06:01:50.5903306Z     [ros2_control_node-2] [INFO] [1780034504.701084421] [controller_manager]: Creating JointSaturationLimiter for joint 'joint2' in hardware 'TestSystemComponent'
2026-05-29T06:01:50.5903853Z     [ros2_control_node-2] [INFO] [1780034504.701422996] [controller_manager]: Using JointLimiter for joint 'joint1' in hardware 'TestSystemComponent' : '  has position limits: true [-3.14159, 3.14159]
2026-05-29T06:01:50.5904067Z     [ros2_control_node-2]   has velocity limits: true [0.2]
2026-05-29T06:01:50.5904226Z     [ros2_control_node-2]   has acceleration limits: false [nan]
2026-05-29T06:01:50.5904381Z     [ros2_control_node-2]   has deceleration limits: false [nan]
2026-05-29T06:01:50.5904578Z     [ros2_control_node-2]   has jerk limits: false [nan]
2026-05-29T06:01:50.5904716Z     [ros2_control_node-2]   has effort limits: true [0.1]
2026-05-29T06:01:50.5904841Z     [ros2_control_node-2]   angle wraparound: false'
2026-05-29T06:01:50.5905296Z     [ros2_control_node-2] [INFO] [1780034504.701446341] [controller_manager]: Creating JointSaturationLimiter for joint 'joint1' in hardware 'TestSystemComponent'
2026-05-29T06:01:50.5905730Z     [ros2_control_node-2] [INFO] [1780034504.703232561] [controller_manager]: Loading hardware 'TestSensorComponent' 
2026-05-29T06:01:50.5906243Z     [ros2_control_node-2] [INFO] [1780034504.704692322] [controller_manager]: Loaded hardware 'TestSensorComponent' from plugin 'test_hardware_components/TestForceTorqueSensor'
2026-05-29T06:01:50.5906598Z     [ros2_control_node-2] [INFO] [1780034504.705132049] [controller_manager]: Initialize hardware 'TestSensorComponent' 
2026-05-29T06:01:50.5906805Z     [ros2_control_node-2] TestForceTorqueSensor configured successfully.
2026-05-29T06:01:50.5907194Z     [ros2_control_node-2] [INFO] [1780034504.717362906] [controller_manager]: Successful initialization of hardware 'TestSensorComponent'
2026-05-29T06:01:50.5907527Z     [ros2_control_node-2] [INFO] [1780034504.717916493] [controller_manager]: Loading hardware 'TestSystemComponent' 
2026-05-29T06:01:50.5908033Z     [ros2_control_node-2] [INFO] [1780034504.718371383] [controller_manager]: Loaded hardware 'TestSystemComponent' from plugin 'test_hardware_components/TestTwoJointSystem'
2026-05-29T06:01:50.5908414Z     [ros2_control_node-2] [INFO] [1780034504.718524512] [controller_manager]: Initialize hardware 'TestSystemComponent' 
2026-05-29T06:01:50.5908599Z     [ros2_control_node-2] TestTwoJointSystem configured successfully.
2026-05-29T06:01:50.5908975Z     [ros2_control_node-2] [INFO] [1780034504.730264145] [controller_manager]: Successful initialization of hardware 'TestSystemComponent'
2026-05-29T06:01:50.5909354Z     [ros2_control_node-2] [INFO] [1780034504.730952263] [controller_manager]: Activating component 'TestSystemComponent'.
2026-05-29T06:01:50.5909678Z     [ros2_control_node-2] [INFO] [1780034504.731049178] [resource_manager]: 'configure' hardware 'TestSystemComponent' 
2026-05-29T06:01:50.5910036Z     [ros2_control_node-2] [INFO] [1780034504.731089468] [resource_manager]: Successful 'configure' of hardware 'TestSystemComponent'
2026-05-29T06:01:50.5910333Z     [ros2_control_node-2] [INFO] [1780034504.731108347] [resource_manager]: 'activate' hardware 'TestSystemComponent' 
2026-05-29T06:01:50.5910692Z     [ros2_control_node-2] [INFO] [1780034504.731132633] [resource_manager]: Successful 'activate' of hardware 'TestSystemComponent'
2026-05-29T06:01:50.5911017Z     [ros2_control_node-2] [INFO] [1780034504.731310809] [controller_manager]: Activating component 'TestSensorComponent'.
2026-05-29T06:01:50.5911318Z     [ros2_control_node-2] [INFO] [1780034504.731334845] [resource_manager]: 'configure' hardware 'TestSensorComponent' 
2026-05-29T06:01:50.5911676Z     [ros2_control_node-2] [INFO] [1780034504.731343007] [resource_manager]: Successful 'configure' of hardware 'TestSensorComponent'
2026-05-29T06:01:50.5911980Z     [ros2_control_node-2] [INFO] [1780034504.731353323] [resource_manager]: 'activate' hardware 'TestSensorComponent' 
2026-05-29T06:01:50.5912316Z     [ros2_control_node-2] [INFO] [1780034504.731361385] [resource_manager]: Successful 'activate' of hardware 'TestSensorComponent'
2026-05-29T06:01:50.5912665Z     [ros2_control_node-2] [INFO] [1780034504.731406442] [controller_manager]: Registering statistics for : TestSystemComponent
2026-05-29T06:01:50.5913071Z     [ros2_control_node-2] [INFO] [1780034504.731582385] [controller_manager]: Registering statistics for : TestSensorComponent
2026-05-29T06:01:50.5913550Z     [ros2_control_node-2] [INFO] [1780034504.731707853] [controller_manager]: Resource Manager has been successfully initialized. Starting Controller Manager services...
2026-05-29T06:01:50.5914429Z     [spawner-3] [INFO] [1780034504.919125729] [spawner_ctrl_with_parameters_and_type]: Setting controller param "params_file" to "['/home/runner/work/ros2_control/ros2_control/.work/target_ws/install/controller_manager/test/test_ros2_control_node.yaml']" for ctrl_with_parameters_and_type
2026-05-29T06:01:50.5915042Z     [spawner-3] [INFO] [1780034504.920848143] [spawner_ctrl_with_parameters_and_type]: Setting controller param "type" to "controller_manager/test_controller" for ctrl_with_parameters_and_type
2026-05-29T06:01:50.5915633Z     [ros2_control_node-2] [INFO] [1780034504.922580743] [controller_manager]: Loading controller : 'ctrl_with_parameters_and_type' of type 'controller_manager/test_controller'
2026-05-29T06:01:50.5915977Z     [ros2_control_node-2] [INFO] [1780034504.922636226] [controller_manager]: Loading controller 'ctrl_with_parameters_and_type'
2026-05-29T06:01:50.5916821Z     [ros2_control_node-2] [INFO] [1780034504.924916533] [controller_manager]: Controller 'ctrl_with_parameters_and_type' node arguments: '--ros-args --params-file /home/runner/work/ros2_control/ros2_control/.work/target_ws/install/controller_manager/test/test_ros2_control_node.yaml'
2026-05-29T06:01:50.5917161Z     [spawner-3] [INFO] [1780034504.947926488] [spawner_ctrl_with_parameters_and_type]: Loaded ctrl_with_parameters_and_type
2026-05-29T06:01:50.5917514Z     [ros2_control_node-2] [INFO] [1780034504.948977959] [controller_manager]: Configuring controller: 'ctrl_with_parameters_and_type'
2026-05-29T06:01:50.5917886Z     [ros2_control_node-2] [INFO] [1780034504.958340901] [controller_manager]: Activating controllers: [ ctrl_with_parameters_and_type ]
2026-05-29T06:01:50.5918306Z     [ros2_control_node-2] [INFO] [1780034504.966525051] [controller_manager]: Requested controller switch from non-realtime loop
2026-05-29T06:01:50.5918599Z     [ros2_control_node-2] [INFO] [1780034504.966691840] [controller_manager]: Successfully switched controllers!
2026-05-29T06:01:50.5919069Z     [spawner-3] [INFO] [1780034504.977879878] [spawner_ctrl_with_parameters_and_type]: Configured and activated ctrl_with_parameters_and_type
2026-05-29T06:01:50.5919259Z     [INFO] [spawner-3]: sending signal 'SIGINT' to process[spawner-3]
2026-05-29T06:01:50.5919490Z     [INFO] [ros2_control_node-2]: sending signal 'SIGINT' to process[ros2_control_node-2]
2026-05-29T06:01:50.5919757Z     [INFO] [robot_state_publisher-1]: sending signal 'SIGINT' to process[robot_state_publisher-1]
2026-05-29T06:01:50.5919981Z     [ros2_control_node-2] [INFO] [1780034505.283857766] [rclcpp]: signal_handler(signum=2)
2026-05-29T06:01:50.5920269Z     [ros2_control_node-2] [INFO] [1780034505.283937174] [controller_manager]: Shutdown request received....
2026-05-29T06:01:50.5920633Z     [ros2_control_node-2] [INFO] [1780034505.283992146] [controller_manager]: Shutting down all controllers in the controller manager.
2026-05-29T06:01:50.5920993Z     [ros2_control_node-2] [INFO] [1780034505.284025396] [controller_manager]: Deactivating controller 'ctrl_with_parameters_and_type'
2026-05-29T06:01:50.5921383Z     [ros2_control_node-2] [INFO] [1780034505.284093568] [controller_manager]: Shutting down controller 'ctrl_with_parameters_and_type'
2026-05-29T06:01:50.5921720Z     [ros2_control_node-2] [INFO] [1780034505.284254408] [resource_manager]: 'deactivate' hardware 'TestSystemComponent' 
2026-05-29T06:01:50.5922077Z     [ros2_control_node-2] [INFO] [1780034505.284276231] [resource_manager]: Successful 'deactivate' of hardware 'TestSystemComponent'
2026-05-29T06:01:50.5922391Z     [ros2_control_node-2] [INFO] [1780034505.284283422] [resource_manager]: 'shutdown' hardware 'TestSystemComponent' 
2026-05-29T06:01:50.5922815Z     [ros2_control_node-2] [INFO] [1780034505.284288329] [resource_manager]: Successful 'shutdown' of hardware 'TestSystemComponent'
2026-05-29T06:01:50.5923122Z     [ros2_control_node-2] [INFO] [1780034505.284388999] [resource_manager]: 'deactivate' hardware 'TestSensorComponent' 
2026-05-29T06:01:50.5923478Z     [ros2_control_node-2] [INFO] [1780034505.284394357] [resource_manager]: Successful 'deactivate' of hardware 'TestSensorComponent'
2026-05-29T06:01:50.5923851Z     [ros2_control_node-2] [INFO] [1780034505.284398013] [resource_manager]: 'shutdown' hardware 'TestSensorComponent' 
2026-05-29T06:01:50.5924239Z     [ros2_control_node-2] [INFO] [1780034505.284402279] [resource_manager]: Successful 'shutdown' of hardware 'TestSensorComponent'
2026-05-29T06:01:50.5924551Z     [ros2_control_node-2] [INFO] [1780034505.284426946] [controller_manager]: Shutting down the controller manager.
2026-05-29T06:01:50.5924806Z     [robot_state_publisher-1] [INFO] [1780034505.287779115] [rclcpp]: signal_handler(signum=2)
2026-05-29T06:01:50.5925111Z     [ros2_control_node-2] [INFO] [1780034505.289167820] [controller_manager.pal_statistics]: Async messages lost 0
2026-05-29T06:01:50.5925435Z     [ros2_control_node-2] [INFO] [1780034505.289181811] [controller_manager.pal_statistics]: publish_async_failures_ 0
2026-05-29T06:01:50.5925927Z     [ros2_control_node-2] [INFO] [1780034505.291789268] [controller_manager.pal_statistics]: Async messages lost 0
2026-05-29T06:01:50.5926231Z     [ros2_control_node-2] [INFO] [1780034505.291803820] [controller_manager.pal_statistics]: publish_async_failures_ 0
2026-05-29T06:01:50.5927520Z     [ERROR] [spawner-3]: process has died [pid 15070, exit code -2, cmd '/home/runner/work/ros2_control/ros2_control/.work/target_ws/install/controller_manager/lib/controller_manager/spawner ctrl_with_parameters_and_type --controller-manager controller_manager --param-file /home/runner/work/ros2_control/ros2_control/.work/target_ws/install/controller_manager/test/test_ros2_control_node.yaml --ros-args'].
2026-05-29T06:01:50.5927812Z     [INFO] [robot_state_publisher-1]: process has finished cleanly [pid 15068]
2026-05-29T06:01:50.5928000Z     [INFO] [ros2_control_node-2]: process has finished cleanly [pid 15069]
2026-05-29T06:01:50.5928202Z     ----------------------------- Captured stderr call -----------------------------
2026-05-29T06:01:50.5928559Z     test_controllers_start (test_ros2_control_node_launch.TestFixture.test_controllers_start) ... ok
2026-05-29T06:01:50.5930043Z     test_node_start (test_ros2_control_node_launch.TestFixture.test_node_start) ... [WARN] [1780034505.156860557] [rcl.logging_rosout]: Publisher already registered for node name: 'test_node'. If this is due to multiple nodes with the same name then all logs for the logger named 'test_node' will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
2026-05-29T06:01:50.5930139Z     ok
2026-05-29T06:01:50.5930238Z     
2026-05-29T06:01:50.5930394Z     ----------------------------------------------------------------------
2026-05-29T06:01:50.5930503Z     Ran 2 tests in 0.680s
2026-05-29T06:01:50.5930591Z     
2026-05-29T06:01:50.5930677Z     OK
2026-05-29T06:01:50.5930898Z     test_exit_codes (test_ros2_control_node_launch.TestShutdown.test_exit_codes)
2026-05-29T06:01:50.5931040Z     Check if the processes exited normally. ... FAIL
2026-05-29T06:01:50.5931132Z     
2026-05-29T06:01:51.3762947Z     ======================================================================
2026-05-29T06:01:51.3763862Z     FAIL: test_exit_codes (test_ros2_control_node_launch.TestShutdown.test_exit_codes)
2026-05-29T06:01:51.3764575Z     Check if the processes exited normally.
2026-05-29T06:01:51.3764967Z     ----------------------------------------------------------------------
2026-05-29T06:01:51.3765341Z     Traceback (most recent call last):
2026-05-29T06:01:51.3766448Z       File "/home/runner/work/ros2_control/ros2_control/.work/target_ws/src/ros2_control/controller_manager/test/test_ros2_control_node_launch.py", line 119, in test_exit_codes
2026-05-29T06:01:51.3767477Z         launch_testing.asserts.assertExitCodes(proc_info)
2026-05-29T06:01:51.3767850Z         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
2026-05-29T06:01:51.3768469Z       File "/opt/ros/lyrical/lib/python3.14/site-packages/launch_testing/asserts/assert_exit_codes.py", line 62, in assertExitCodes
2026-05-29T06:01:51.3769317Z         assert info.returncode in allowable_exit_codes, 'Proc {} exited with code {}'.format(
2026-05-29T06:01:51.3769774Z                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-05-29T06:01:51.3770106Z     AssertionError: Proc spawner-3 exited with code -2
2026-05-29T06:01:51.3770403Z     
2026-05-29T06:01:51.3770672Z     ----------------------------------------------------------------------
2026-05-29T06:01:51.3771025Z     Ran 1 test in 0.001s
2026-05-29T06:01:51.3771267Z     
2026-05-29T06:01:51.3771471Z     FAILED (failures=1)
2026-05-29T06:01:51.3772183Z     - generated xml file: /home/runner/work/ros2_control/ros2_control/.work/target_ws/build/controller_manager/test_results/controller_manager/test_ros2_control_node.xunit.xml -
2026-05-29T06:01:51.3772963Z     =========================== short test summary info ============================
2026-05-29T06:01:51.3773614Z     FAILED ../../src/ros2_control/controller_manager/test/test_ros2_control_node_launch.py::test_ros2_control_node_launch
2026-05-29T06:01:51.3774232Z     ============================== 1 failed in 1.27s ===============================
2026-05-29T06:01:51.3774594Z     -- run_test.py: return code 1
2026-05-29T06:01:51.3775344Z     -- run_test.py: verify result file '/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/controller_manager/test_results/controller_manager/test_ros2_control_node.xunit.xml'
2026-05-29T06:01:51.3776331Z   >>>
```

--- a/controller_manager/test/test_ros2_control_node_launch.py
+++ b/controller_manager/test/test_ros2_control_node_launch.py
@@ -105,9 +105,17 @@ class TestFixture(unittest.TestCase):
     def test_node_start(self):
         check_node_running(self.node, "controller_manager")
 
-    def test_controllers_start(self):
+    def test_controllers_start(self, proc_info):
         cnames = ["ctrl_with_parameters_and_type"]
-        check_controllers_running(self.node, cnames, state="active")
+
+        check_controllers_running(self.node, cnames)
+
+        # Wait for controller_spawner to finish and verify successful exit.
+        proc_info.assertWaitForShutdown(process="spawner", timeout=30)
+        launch_testing.asserts.assertExitCodes(proc_info, process="spawner")
+
+        # Re-check controllers after spawner has exited.
+        check_controllers_running(self.node, cnames)
 
 
 @launch_testing.post_shutdown_test()

